### PR TITLE
Detect globalnet in subctl verify

### PR DESCRIPTION
**Pull request by tpantelis**
Read the operator Submariner resource from one of the clusters and
check the GlobalCIDR field.

Signed-off-by: Tom Pantelis <tompantelis@gmail.com>

### CODE REVIEW

File: pkg/subctl/cmd/verify.go [hunk 1]
```diff
@@ -264,3 +268,18 @@ func getVerifyPatterns(csv string, includeDisruptive bool) ([]string, []string,
 	}
 	return outputPatterns, outputVerifications, nil
 }
+
+func detectGlobalnet() {
+	submarinerClient, err := submarinerclientset.NewForConfig(framework.RestConfigs[framework.ClusterA])
+	if err != nil {
+		exitWithErrorMsg(fmt.Sprintf("Error creating submariner client: %v", err))
+	}
```
Comment (mangelajo) [inline pkg/subctl/cmd/verify.go:276]:
```suggestion
	exitOnError("Error creating submariner client", err)
```

File: pkg/subctl/cmd/verify.go [hunk 2]
```diff
@@ -264,3 +268,18 @@ func getVerifyPatterns(csv string, includeDisruptive bool) ([]string, []string,
...
+
+	submariner, err := submarinerClient.SubmarinerV1alpha1().Submariners(OperatorNamespace).Get(submarinercr.SubmarinerName,
+		v1.GetOptions{})
+	if err != nil {
+		exitWithErrorMsg(fmt.Sprintf("Error obtaining Submariner resource: %v", err))
+	}
```
Comment (mangelajo) [inline pkg/subctl/cmd/verify.go:282]:
```suggestion
	exitOnError("Error obtaining Submariner resource", err)
```

File: pkg/subctl/cmd/verify.go (hunk 1 above)
Comment (tpantelis) [reply pkg/subctl/cmd/verify.go]: Done

File: pkg/subctl/cmd/verify.go (hunk 2 above)
Comment (tpantelis) [reply pkg/subctl/cmd/verify.go]: Done

File: pkg/subctl/cmd/verify.go [hunk 3]
```diff
@@ -264,3 +268,14 @@ func getVerifyPatterns(csv string, includeDisruptive bool) ([]string, []string,
 	}
 	return outputPatterns, outputVerifications, nil
 }
+
+func detectGlobalnet() {
+	submarinerClient, err := submarinerclientset.NewForConfig(framework.RestConfigs[framework.ClusterA])
+	exitWithErrorMsg(fmt.Sprintf("Error creating submariner client: %v", err))
```
Comment (mangelajo) [inline pkg/subctl/cmd/verify.go:274]:
```suggestion
	exitOnError("Error creating submariner client", err)
```

File: pkg/subctl/cmd/verify.go [hunk 4]
```diff
@@ -264,3 +268,14 @@ func getVerifyPatterns(csv string, includeDisruptive bool) ([]string, []string,
...
+
+	submariner, err := submarinerClient.SubmarinerV1alpha1().Submariners(OperatorNamespace).Get(submarinercr.SubmarinerName,
+		v1.GetOptions{})
+	exitWithErrorMsg(fmt.Sprintf("Error obtaining Submariner resource: %v", err))
```
Comment (mangelajo) [inline pkg/subctl/cmd/verify.go:278]:
```suggestion
	exitOnError("Error obtaining Submariner resource", err))
```

**Comment by mangelajo** [comment]
Thanks tom

**Comment by tpantelis** [comment]
> This means `subctl verify` can only be run against operator deployments. Which should be fine, but need to add a check and warning/error msg somewhere if someone runs it against a non-operator deployment. Or at least capture it in documentation. Maybe raise an issue that we can close once helm deployments move to operator underneath.

`subctl` is part of the operator project and the most important part of it, `join`, relies on the operator so I think it's reasonable for `verify` to do so. It'll fail if the `Submariner` object isn't found. I can improve the error message to make it clear that the operator must be deployed and helm deployment is not supported.

**Comment by tpantelis** [comment]
> I can improve the error message to make it clear that the operator must be deployed and helm deployment is not supported.

I updated the error message via https://github.com/submariner-io/submariner-operator/pull/566/commits/3e375a67f93a8e3bde685070f8c40cf0f0b402bd

**Pull request merged**